pkg/lib/synthesize/non_inferrable_csv.go

```diff
@@ -0,0 +1,31 @@
+package synthesize
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/operator-framework/api/pkg/lib/version"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+type RegistryV2CSV struct {
+	Name                      string                           `json:"name,omitempty"`
+	GenerateName              string                           `json:"generateName,omitempty"`
+	Kind                      string                           `json:"kind"`
+	MinKubeVersion            string                           `json:"minKubeVersion,omitempty"`
+	InstallModes              []v1alpha1.InstallMode           `json:"installModes,omitempty"`
+	Description               string                           `json:"description,omitempty"`
+	DisplayName               string                           `json:"displayName,omitempty"`
+	Keywords                  []string                         `json:"keywords,omitempty"`
+	Labels                    map[string]string                `json:"labels,omitempty" `
+	Links                     []v1alpha1.AppLink               `json:"links,omitempty"`
+	Maintainers               []v1alpha1.Maintainer            `json:"maintainers,omitempty"`
+	Maturity                  string                           `json:"maturity,omitempty"`
+	Provider                  v1alpha1.AppLink                 `json:"provider,omitempty"`
+	Selector                  *metav1.LabelSelector            `json:"selector,omitempty"`
+	Icon                      []v1alpha1.Icon                  `json:"icon,omitempty"`
+	Replaces                  string                           `json:"replaces,omitempty"`
+	Version                   version.OperatorVersion          `json:"version,omitempty"`
+	Annotations               map[string]string                `json:"annotations,omitempty"`
+	CustomResourceDefinitions []v1alpha1.CRDDescription        `json:"customresourcedefinitions,omitempty"`
+	APIServiceDefinitions     []v1alpha1.APIServiceDescription `json:"apiservicedefinitions,omitempty"`
```

Comment:
We may be able to avoid surfacing these if we merge your [PR first](https://github.com/operator-framework/operator-lifecycle-manager/pull/1435), which allows us to treat all requirements as native APIs. Once that's in, we can generate the NativeAPIs field on the output CSV [with info from the `dependencies.yaml` file](https://github.com/operator-framework/operator-registry/blob/8a8da5b42668a49fa1ff6e54565bb9cc644f00a2/pkg/registry/types.go#L138).

Reply:
I don't think we should remove these. They allow users to retain the ability to describe owned/required CRD, APIService, and Service. Unless we do not want any description anymore, which i thought would be a problem for UI?